packages/llamaindex/src/storage/vectorStore/MongoDBAtlasVectorStore.ts

```diff
@@ -244,4 +244,26 @@ export class MongoDBAtlasVectorSearch
     console.debug("Result of query (ids):", ids);
     return result;
   }
+
+  async createSearchIndex() {
```

Comment:
how about
1. create this.collection lazily (async funciton)
2. add `autoCreateIndex` flag to constructor (defaults to `true`)
3. if `autoCreateIndex==true`, create the index (if necessary)

then user doesn't have to call `vectorStore.createSearchIndex();`

(better DX)

- also don't forget to about README (removing manually creating the index)